tests/runner/src/main.rs

```diff
@@ -67,16 +67,22 @@ struct Args {
     /// If not set, sqlness will build GreptimeDB on the fly.
     #[clap(long)]
     bins_dir: Option<PathBuf>,
+
+    /// Preserve persistent state in the temporary directory.
+    /// This may affect future test runs.
+    #[clap(long)]
+    preserve_state: bool,
 }
 
 #[tokio::main]
 async fn main() {
     let args = Args::parse();
 
-    #[cfg(windows)]
-    let data_home = std::env::temp_dir();
-    #[cfg(not(windows))]
-    let data_home = std::path::PathBuf::from("/tmp");
+    let temp_dir = tempfile::Builder::new()
+        .prefix("sqlness")
+        .tempdir()
+        .unwrap();
+    let data_home = temp_dir.path().to_path_buf();
```

Comment:
The compatibility test failed because of this change. Here sqlness uses a new temp dir everytime while the compatibility test demands a consistent data view from last run. I suggest adding a new argument called "data_home" in this sqlness runner, and set a fixed dir in "tests/compat/util.sh".

Reply:
I know this is the reason. The problem is I think this is a misuse of sqlness' bug which doesn't clear DB's status.

For the integration test, I don't think reuse sqlness is the best choice. Sqlness case is always up-to-date and contains the newest feature. Hence it's not possible to test forward compatibility. That's the reason I think we need a dedicate script for both forward and backward compatibility -- at least sqlness cannot distinguish features with version.

Reply:
I conclude my opinion at https://github.com/GreptimeTeam/greptimedb/issues/3728, please check it out

Reply:
I'm ok with #3728, but before we arrive at that destination, at least we have a viable option to test compatibility here.

Reply:
>at least we have a viable option to test compatibility here.

We still can run it manually, only the automation script is outdated.

Reply:
Manually running the compatibility test makes it hard to find any accidently breaking changes across each PR comparing to run it in the CI. I think if there are no hard limits that sqlness cannot be used here, we should fix it now. Then try the better solution described in #3728. WDYT @killme2008 @fengjiachun

Reply:
I'll add a naive implement for #3728 to replace the current misuse of sqlness, let's back to this patch after that.

Reply:
Ok, let's go ahead.